server/routes/api/representatives.js

```diff
@@ -50,15 +50,32 @@ const ALLOWED_JURISDICTION_FILTERS = Object.keys(JURISDICTION_FILTER_MAP)
 // Endpoints
 
 // Get
-router.get('/:zipCode', async (req, res) => {
-  const { zipCode } = req.params
+router.get('/:searchText', async (req, res) => {
+  const { searchText } = req.params
   const { filter } = req.query
 
-  if (!zipCode.match(/^\d{5}(-\d{4})?$/)) {
+  /* check if valid ZIP code 
+    ^\d{5} - the start of the line has 5 digits
+    (-\d{4})?$ the end of the line has 4 digits preceded by a dash (optionally)
+  */
+  let isValidZIPcode = /^\d{5}(-\d{4})?$/.test(searchText)
+  /* check if valid street address
+     ^(\d{3,})\s? the start of the line can have 3 or more digits followed by a space
+     (\w{0,})\s the next part can have 2 or more letters followed by a space
+     ([a-zA-Z]{2,30})\s the next part must be an alphanetical string with 2-30 characters followed by a space
+     ([a-zA-Z]{2,15})\.?\s?  the next part must be an alphabetical string with 2-15 characters followed by a any digit (optionally) and/or space (optionally)
+     (\w{0,5})$ the end of the line can have 0-5 letters
+  */
+  let isValidAddress =
+    /^(\d{3,})\s?(\w{0,5})\s([a-zA-Z]{2,30})\s([a-zA-Z]{2,15})\.?\s?(\w{0,5})$/.test(
```

Comment:
The regex is probably more robust if it requires a space (`+`) instead of allowing one or zero (`?`)

```suggestion
    /^(\d{3,})\s+(\w{0,5})\s+([a-zA-Z]{2,30})\s+([a-zA-Z]{2,15})\.?\s+(\w{0,5})$/.test(
```

Reply:
@thyeggman 
I do not agree with this. Some sections should be optional, and this regex should also target several cases. I am not as familiar with US addresses, but as far as I know, these are all valid:
- 101 Main St N
- 101 W Main St 
- 123 45th street (you used this as an example for another suggestion).

But they would not work with the new suggestion.